ansible/roles/local_storage/templates/localvolume.yml.j2

```diff
@@ -8,15 +9,16 @@ spec:
   nodeSelector: 
     nodeSelectorTerms:
     - matchExpressions:
-      - key: "{{local_storage_instances.key}}"
+      - key: "{{instances.key}}"
         operator: In
         values:
-        - "{{local_storage_instances.value}}"
+        - "{{instance.value}}"
   storageClassDevices:
-  - storageClassName: "local-sc"
+  - storageClassName: "{{instance.storageclassname}}"
     volumeMode: Filesystem 
     fsType: xfs 
     devicePaths:
-{% for volume in local_storage_instances.volumes %}
+  {% for volume in instance.volumes %}
       - "{{volume}}"
+  {% endfor %}
```

Comment:
I also had a silly whitespace issue here.  Since I wasn't able to discern the correct jinja2 syntax to trim only the whitespace I wanted, I removed the whitespace entirely by doing indents inside the `{%` jinja brackets.  It looks like this instead now:
```
...
    volumeMode: Filesystem 
    fsType: xfs 
    devicePaths:
{%   for volume in instance.volumes %}
      - "{{volume}}"
{%   endfor %}
{% endfor %}
```
With that my whitespace issue was no more.